### PR TITLE
Bump pyyaml from 5.3.1 to 5.4.1

### PR DESCRIPTION
**Pull request by qiluo-msft**
**- Why I did it**
RCE resolved in new version https://github.com/yaml/pyyaml/issues/420

**- How I did it**

**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by jleveque — CHANGES_REQUESTED** [review]
As part of this PR, can you please also pin the PyYAML version in:

- src/sonic-ctrmgrd/setup.py
- src/sonic-py-common/setup.py

And determine if we still need to explicitly install it in dockers/docker-snmp/Dockerfile.j2?

File: sonic-slave-buster/Dockerfile.j2 [hunk 1]
```diff
@@ -411,8 +411,8 @@ RUN pip3 uninstall -y enum34
 RUN pip2 install j2cli==0.3.10
 
 # For sonic-mgmt-framework
-RUN pip2 install "PyYAML==5.3.1"
-RUN pip3 install "PyYAML==5.3.1"
+RUN pip2 install "PyYAML==5.4.1"
+RUN pip3 install "PyYAML==5.4.1"
```
[the remark this answers is not in the document]
Comment (qiluo-msft) [reply sonic-slave-buster/Dockerfile.j2:415]: As the comment say "For sonic-mgmt-framework", I already talked the owner and there is no easy way to specify this build dependency in that repo.
ref: [https://github.com/Azure/sonic-mgmt-framework/issues/78](https://github.com/Azure/sonic-mgmt-framework/issues/78)

---
In reply to: [561603081](https://github.com/Azure/sonic-buildimage/pull/6511#discussion_r561603081) [](ancestors = 561603081)

File: sonic-slave-stretch/Dockerfile.j2 [hunk 2]
```diff
@@ -346,8 +346,8 @@ RUN pip2 install j2cli==0.3.10
 RUN pip3 install mockredispy==2.9.3
 
 # For sonic-mgmt-framework
-RUN pip2 install "PyYAML==5.3.1"
-RUN pip3 install "PyYAML==5.3.1"
+RUN pip2 install "PyYAML==5.4.1"
+RUN pip3 install "PyYAML==5.4.1"
```
[the remark this answers is not in the document]
Comment (qiluo-msft) [reply sonic-slave-stretch/Dockerfile.j2:350]: The same

---
In reply to: [561603094](https://github.com/Azure/sonic-buildimage/pull/6511#discussion_r561603094) [](ancestors = 561603094)

**Comment by qiluo-msft** [comment]
Both are good now. Even the dependency is not available in the build environment, the downloaded latest version is good. If they need any specific version, they need to pin it.

---
In reply to: [572984163](https://github.com/Azure/sonic-buildimage/pull/6511#issuecomment-572984163) [](ancestors = 572984163)

**Comment by qiluo-msft** [comment]
Retest this please

**Comment by qiluo-msft** [comment]
Retest this please

**Comment by qiluo-msft** [comment]
Retest baseimage please

File: sonic-slave-buster/Dockerfile.j2 (hunk 1 above)
[the remark this answers is not in the document]
Comment (jleveque) [reply sonic-slave-buster/Dockerfile.j2:433]: That's unfortunate. Will need to investigate this further in the future.

**Review by jleveque — APPROVED** [review]

**Pull request merged**